Sources/SwiftSyntaxBuilder/gyb_helpers/BuilderInitializableTypes.py

```diff
@@ -0,0 +1,15 @@
+BUILDER_INITIALIZABLE_TYPES = {
+    'CodeBlock': 'CodeBlockItemList',
+    'MemberDeclBlock': 'MemberDeclList',
+    'CodeBlockItemList': None,
+    'MemberDeclList': None,
+    'PatternBindingList': None,
+    'SwitchCaseList': None,
+    'ArrayElementList': None,
+    'TupleExprElementList': None,
+    'EnumCaseElementList': None,
+    'FunctionParameterList': None,
+    'GenericParameterList': None,
+    'GenericRequirementList': None,
+    'InheritedTypeList': None,
```

Comment:
I have added quite a few new types here, mostly since the array-based initializers do not support the new trailing comma logic that was recently merged. In general my suggestion would be that we try to use builders for every parameter that can be reasonably well expressed using a single trailing closure, since that usually produces very concise and nice-looking DSL code.

Reply:
That also plays well e.g. with function parameter lists generated by for-loops (within a builder)

Reply:
Another thing I thought about was whether we would want to automatically create builders for every `HasTrailingComma` type. This would e.g. in `IfStmt`, however, lead to an initializer with multiple builder closures (`conditions` and `body`), which we'd like to avoid, therefore I went with the manual approach for now.

Reply:
I agree that maintaining this list manually is the better option because, as you mentioned, it gives us the flexibility to try and avoid initializers that take multiple result builders.